### DISCUSSION TITLE
[V5] Complete Event callback.

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#3046] [Help] [answered]
Hi Team,
Can we use event onComplete callback in adonis event ?
If yes please share a code snap.
Thanks,
@thetutlage

**Comment by McSneaky** [comment, marked as answer]
AFAIK there's no such thing, but you can use `.then()`

```ts
Event.emit('myEvent', payload)
  .then(() => {
    console.log('My event completed')
  })
```

Or emit another event in end of your event, but that might go real spagetty real fast

**Reply by pk4all** [reply]
Thanks.
